code/modules/library/computers/checkout.dm

```diff
@@ -0,0 +1,465 @@
+			else if(!scanner.cache)
+				dat += "<FONT color=red>No data found in scanner memory.</FONT><BR>"
+			else
+
+				dat += {"<TT>Data marked for upload...</TT><BR>
+					<TT>Title: </TT>[scanner.cache.name]<BR>"}
+				if(!scanner.cache.author)
+					scanner.cache.author = "Anonymous"
+
+				dat += {"<TT>Author: </TT><A href='?src=\ref[src];uploadauthor=1'>[scanner.cache.author]</A><BR>
+					<TT>Category: </TT><A href='?src=\ref[src];uploadcategory=1'>[upload_category]</A><BR>
+					<A href='?src=\ref[src];upload=1'>\[Upload\]</A><BR>"}
+			dat += "<A href='?src=\ref[src];switchscreen=0'>(Return to main menu)</A><BR>"
+		if(7)
+			dat += "<H3>Print a Manual</H3>"
+			dat += "<table>"
+
+			var/list/forbidden = list(
+				/obj/item/weapon/book/manual
+			)
+
+			if(!emagged)
+				forbidden |= /obj/item/weapon/book/manual/nuclear
+
+			var/manualcount = 1
+			var/obj/item/weapon/book/manual/M = null
+
+			for(var/manual_type in (typesof(/obj/item/weapon/book/manual) - forbidden))
+				M = new manual_type()
+				dat += "<tr><td><A href='?src=\ref[src];manual=[manualcount]'>[M.title]</A></td></tr>"
+				manualcount++
+				qdel(M)
+				M = null
+			dat += "</table>"
+			dat += "<BR><A href='?src=\ref[src];switchscreen=0'>(Return to main menu)</A><BR>"
+
+		if(8)
+
+			dat += {"<h3>Accessing Forbidden Lore Vault v 1.3</h3>
+				Are you absolutely sure you want to proceed? EldritchTomes Inc. takes no responsibilities for loss of sanity resulting from this action.<p>
+				<A href='?src=\ref[src];arccheckout=1'>Yes.</A><BR>
+				<A href='?src=\ref[src];switchscreen=0'>No.</A><BR>"}
+
+	var/datum/browser/B = new /datum/browser(user, "library", "Book Inventory Management")
+	B.set_content(dat)
+	B.open()
+
+/obj/machinery/computer/library/checkout/emag_act(mob/user)
+	if(density && !emagged)
+		emagged = 1
+		to_chat(user, "<span class='notice'>You override the library computer's printing restrictions.</span>")
+
+/obj/machinery/computer/library/checkout/attackby(obj/item/weapon/W as obj, mob/user as mob)
+	if(istype(W, /obj/item/weapon/barcodescanner))
+		var/obj/item/weapon/barcodescanner/scanner = W
+		scanner.computer = src
+		to_chat(user, "[scanner]'s associated machine has been set to [src].")
+		audible_message("[src] lets out a low, short blip.", 2)
+		return 1
+	else
+		return ..()
+
+/obj/machinery/computer/library/checkout/Topic(href, href_list)
+	if(..())
+		usr << browse(null, "window=library")
+		onclose(usr, "library")
+		return 1
+
+	if(href_list["pagenum"])
+		if(!num_pages)
+			page_num = 1
+		else
+			var/pn = text2num(href_list["pagenum"])
+			if(!isnull(pn))
+				page_num = Clamp(pn, 1, num_pages)
+
+	if(href_list["page"])
+		if(num_pages == 0)
+			page_num = 1
+		else
+			page_num = Clamp(text2num(href_list["page"]), 1, num_pages)
+	if(href_list["settitle"])
+		var/newtitle = input("Enter a title to search for:") as text|null
+		if(newtitle)
+			query.title = sanitize(newtitle)
+		else
```

Comment:
<a href=https://github.com/ParadiseSS13/Paradise/blob/master/code/__HELPERS/text.dm#L40-L43>`sanitize` only filters out "\n" and "\t" by default.</a>. Since these are directly passed to an SQL query, you may want to do `sanitizeSQL` instead.